web/packages/teleport/src/Discover/Server/DiscoveryConfigSsm/DiscoveryConfigSsm.test.tsx

```diff
@@ -0,0 +1,161 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Info } from 'design/Alert';
+import { fireEvent, render, screen, waitFor } from 'design/utils/testing';
+
+import cfg from 'teleport/config';
+import {
+  RequiredDiscoverProviders,
+  resourceSpecAwsEc2Ssm,
+} from 'teleport/Discover/Fixtures/fixtures';
+import { AgentMeta, AutoDiscovery } from 'teleport/Discover/useDiscover';
+import * as discoveryService from 'teleport/services/discovery/discovery';
+import {
+  IntegrationKind,
+  IntegrationStatusCode,
+} from 'teleport/services/integrations';
+import JoinTokenService from 'teleport/services/joinToken';
+import TeleportContext from 'teleport/teleportContext';
+
+import { DiscoveryConfigSsm } from './DiscoveryConfigSsm';
+
+function completeForm() {
+  expect(
+    screen.getByText('Setup Discovery Config for Teleport Discovery Service')
+  ).toBeInTheDocument();
+  let selectEl = screen.getByLabelText(/aws region/i);
+  fireEvent.focus(selectEl);
+  fireEvent.keyDown(selectEl, { key: 'ArrowDown' });
+  fireEvent.click(screen.getByText('us-east-2'));
+  fireEvent.click(screen.getByTestId('region-next'));
+  fireEvent.click(screen.getByTestId('script-next'));
+  expect(
+    screen.getByText(/You can filter for EC2 instances by their tags/i)
+  ).toBeInTheDocument();
+}
+
+async function submitConfig() {
+  fireEvent.click(screen.getByTestId('action-next'));
+  await waitFor(() => {
+    expect(screen.getByTestId('action-next')).toBeDisabled();
+  });
+  expect(discoveryService.createDiscoveryConfig).toHaveBeenCalledTimes(1);
+}
+
+const defaultIsCloud = cfg.isCloud;
+
+beforeAll(() => {
+  cfg.isCloud = true;
+});
+beforeEach(() => {
+  jest.spyOn(discoveryService, 'createDiscoveryConfig');
+  jest
+    .spyOn(JoinTokenService.prototype, 'fetchJoinTokenV2')
+    .mockResolvedValue(tokenResp);
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+afterAll(() => {
+  cfg.isCloud = defaultIsCloud;
+});
+
+test('calls with wildcard tag', async () => {
+  const ctx = new TeleportContext();
+
+  render(<Component ctx={ctx} />);
+  completeForm();
+  await submitConfig();
+  const [[, configArg]] = (discoveryService.createDiscoveryConfig as jest.Mock)
+    .mock.calls;
+  expect(configArg.aws[0].tags).toEqual({ '*': ['*'] });
+});
+
+test('calls with entered tags', async () => {
+  const ctx = new TeleportContext();
+
+  render(<Component ctx={ctx} />);
+  completeForm();
+  // add tags
+
+  fireEvent.click(screen.getByText(/add a tag/i));
+  const keyInput = screen.getByPlaceholderText('label key');
+  const valInput = screen.getByPlaceholderText('label value');
+
+  fireEvent.change(keyInput, { target: { value: 'asdf' } });
+  fireEvent.change(valInput, { target: { value: 'fdsa' } });
+
+  await submitConfig();
+  const [[, configArg]] = (discoveryService.createDiscoveryConfig as jest.Mock)
+    .mock.calls;
+  expect(configArg.aws[0].tags).toEqual({ asdf: ['fdsa'] });
```

Comment:
can we also add a test for `same key, multi value` as well?

Reply:
actually, we can add multivalue to the API call and still use the "single key:value" as the input (if a duplicate key is added, append the values into a set that is returned with `makeTags`). ill do that instead and add tests